sklearn/utils/estimator_checks.py

```diff
@@ -238,17 +247,30 @@ def check_estimator(Estimator):
     will be run if the Estimator class inherits from the corresponding mixin
     from sklearn.base.
 
+    This test can be applied to classes or instances.
+    Classes currently have some additional tests that related to construction,
+    while passing instances allows the testing of multiple options.
+
     Parameters
     ----------
     Estimator : class
-        Class to check. Estimator is a class object (not an instance).
+        Estimator to check. Estimator is a class object or instance.
 
     """
-    name = Estimator.__name__
-    check_parameters_default_constructible(name, Estimator)
-    for check in _yield_all_checks(name, Estimator):
+    if isinstance(Estimator, type):
```

Comment:
I'm not entirely certain this is the right check, but my attempts to break it have failed so far.